src/utils.ts

```diff
@@ -8,12 +10,20 @@ export function debounce(fn: Function, n = 100) {
   }
 }
 
-// prefer old unicode hacks for backward compatibility
-// https://base64.guru/developers/javascript/examples/unicode-strings
 export function utoa(data: string): string {
-  return btoa(unescape(encodeURIComponent(data)))
+  const zipped = deflate(data, { level: 9 })
+  const b = Array.from(zipped, v => String.fromCharCode(v)).join('')
+  return btoa(b)
 }
 
 export function atou(base64: string): string {
-  return decodeURIComponent(escape(atob(base64)))
+  const b = atob(base64)
+  if (b.startsWith('\x78\xDA')) {
+    const buffer = Uint8Array.from(b, (_, i) => b.charCodeAt(i))
+    return inflate(buffer, { to: 'string' })
+  }
+
+  // old unicode hacks for backward compatibility
+  // https://base64.guru/developers/javascript/examples/unicode-strings
+  return decodeURIComponent(escape(b))
```

Comment:
is it necessary to decode if the string wasn't initially encoded?

Reply:
This is so it still supports old links. Line 21 is checking for the ZLIB header.

Reply:
ah of course 🤦‍♂️